src/components/Board.js

```diff
@@ -4,19 +4,23 @@ import Square from './Square';
 import PropTypes from 'prop-types';
 
 
-const generateSquareComponents = (squares, onClickCallback) => {
-  // Complete this for Wave 1
-  // squares is a 2D Array, but 
-  //  you need to return a 1D array
-  //  of square components
+const generateSquareComponents = (squares, onClickCallback) => { // TODO: Is this parameter even needed?
+  const flattened = [];
+  for (let row = 0; row < 3; row += 1) {
+    for (let col = 0; col < 3; col +=1) {
+      flattened.push(squares[row][col]);
+    }
+  }
+
+  return flattened;
```

Comment:
Currently, this function makes a 1D copy of the `squares` array. What you could do instead is make `flattened` a list of Square components that you'll render as {squareList} below.  
```suggestion
const generateSquareComponents = (squares, onClickCallback) => { // TODO: Is this parameter even needed?
  const flattened = [];
  for (let row = 0; row < 3; row += 1) {
    for (let col = 0; col < 3; col +=1) {
      flattened.push(<Square key={square.id} id={square.id} value={square.value} onClickCallback={onClickCallback} />);
    }
  }

  return flattened;
```